### DISCUSSION TITLE
How to override components prop types?

### DISCUSSION THREAD

**Discussion by BOHDANff** [mantinedev/mantine#5615] [Q&A] [answered]
Hello, how to override components prop types? For example I need to override type of Button variant prop to make it more strict (I mean that now variant prop has type `string | ButtonVariant` and I want to make it only like `ButtonVariant` without `string`). Also I want to add some custom variants to the type. So how could I rewrite Button variant prop type?

**Comment by rtivital** [comment]
```tsx
interface MyButtonProps extends Omit<ButtonProps, 'variant'> {
  variant: ButtonVariant;
}
```

**Reply by BOHDANff** [reply]
Thank you for the answer. But if choose this way the button will stop to be polymorphic [![Edit broken-wind-ls9cvh](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/broken-wind-ls9cvh?file=%2Fsrc%2FApp.tsx%3A14%2C32)

**Reply by rtivital** [reply, marked as answer]
https://mantine.dev/guides/polymorphic/#create-your-own-polymorphic-components

**Reply by BOHDANff** [reply]
Thank you, that is what i need
